### ISSUE TITLE
Sybil Attacker Report

### ISSUE THREAD

**Issue by GreyPK** [hop-protocol/hop-airdrop#194]
# Related Addresses

29 related addresses. 
```
0x5a49271faa9c068dc409f6ee52cd260971f213b3
0x469748c862cec0a18b6dc10be7484932ba0d84ca
0x8d01fd620ba95d3e38879206b2a8fbec24885b10
0x1c568afbd66f1669b9870cd33c83a6a5de10dd00
0x95dd72e695017c690bdfa7bca821fc2c5df927af
0xa9bbbead5d03e71634870c7bb7e085d8db64e08c
0xe7fc210397d8f654c064adf4712314c7e7c5cd17
0xdbf199ba9617d5f92d9ed0fb28a2f5f7244df836
0x70b62b87c490bd492f788d9e29ed8d367a68b9aa
0x468a1ce5593ad1dcdabd3357d31eaa95590e35f1
0x61dda79dcb2186893bbc6456de302688c84fd978
0xc56ee7d507e6b20c867f14916d761d04d0a23450
0x1791f1e9cf7e7bdcc0147b466b8b9c797db25a74
0xc46c66313b19189a54cfa5216fa00d7b868a0b50
0x5df2a53369e688000a726da7fc6fcdbd0a3c5102
0xe4faca92515ea49041f5257eae5fff242c7df385
0x914bc451407f9e27f9914c037e88bf545bec0d8a
0xeda6de28eef7df0d2c7bba45b9d068e85c2e89e0
0xb1966b52deca0d4b2c9e23977563b948da3b8360
0x6cda151ad03adabdddf8fa15ffbbe7dc1b8071fa
0xc2fed74f6e44333e1f90becd98c1f41dd31299b7
0x3c32391d71f2bda88941126b2c68fb49a874ccee
0x0d10a6fdf0407580755ed4e9444efa016adf9263
0x37980d2ec65676fee74f6a6e34d1125825c0297e
0xfe2b718a9b26be75a67f444619714b8c5d4f6f39
0xc8442d72e06cd03e5a5b7a01a4dce99db5540feb
0x8983d0d5a892865339e48738f77d37ba9be32df7
0x7b91b1da16d1700120948eb24aee40f058dff032
0x653e791dcec2d311848742b9ad35ca68acfe4839
```

# Reasoning

1) An elligible for the airdrop Address 0x5a49271faa9c068dc409f6ee52cd260971f213b3 send MATIC to these 28 addresses that elligible for the airdrop. These transactions can find here: https://polygonscan.com/txs?a=0x5a49271faa9c068dc409f6ee52cd260971f213b3&f=2&ps=100&p=2
2) All 29 wallets first used HOP with SwapAndSend in almost same day (19 Nov 2021 or 11 Dec 2021  or 14 Dec 2021).
3) At 11 Dec 2021 always swapAndSend 200-250 USDC, At 14 Dec 2021 always swapAndSend 200-250 USDC, At 19 Nov 2021 always swapAndSend 150-220 USDT.

Sender 0x5a49271faa9c068dc409f6ee52cd260971f213b3 Dec-14-2021 used HOP swapAndSend 213 USDC
https://blockscout.com/xdai/mainnet/tx/0x14a745b94b9944ebcc026d8c949328670d63eba633188330395a6557c481ad1c

0x469748c862cec0a18b6dc10be7484932ba0d84ca Dec-11-2021 used HOP swapAndSend 246 USDC
https://polygonscan.com/tx/0x491f5ec433fdb03b476c25c8a0e6c2621bbe133ecc2309fc708f0fcba14cc302

0x8d01fd620ba95d3e38879206b2a8fbec24885b10 Dec-11-2021 used HOP swapAndSend 237 USDC
https://polygonscan.com/tx/0x558f9b889308acf3f5075e58ce2e052bd66ccd49de949dc10d8cb6136e73d6a3

0x1c568afbd66f1669b9870cd33c83a6a5de10dd00 Dec-11-2021 used HOP swapAndSend 244 USDC
https://polygonscan.com/tx/0xc7a12d4b5380105089d665eedd711809254e4c77ec3623184efa4968c33a0a3d

0x95dd72e695017c690bdfa7bca821fc2c5df927af Dec-11-2021 used HOP swapAndSend 240 USDC
https://polygonscan.com/tx/0xab46763527c6457629cb219e5b3bf2148beb9701ea7a684a6a201240db98ce73

0xa9bbbead5d03e71634870c7bb7e085d8db64e08c Dec-11-2021 used HOP swapAndSend 229 USDC
https://polygonscan.com/tx/0x7f49478a423fdcdf0f7368a8bbeedf85ceb16d7fdc5e8a89774f7283ba7d877f

0xe7fc210397d8f654c064adf4712314c7e7c5cd17 Nov-19-2021 used HOP swapAndSend 127 USDT
https://polygonscan.com/tx/0x3023fdb378217d49d94a32ef2536c781c7b8c48d5710b61fc38979026ce165fc

0xdbf199ba9617d5f92d9ed0fb28a2f5f7244df836 Nov-19-2021 used HOP swapAndSend 118 USDT
https://polygonscan.com/tx/0x692b1da87086b915b7f7d9330c92d5d8e59e24b29fd9672a48c9477279d06a01

0x70b62b87c490bd492f788d9e29ed8d367a68b9aa Nov-19-2021 used HOP swapAndSend 243 USDT
https://polygonscan.com/tx/0xe4d5a748fdf6ebc855086ac4022f648966082f2708c35488b466872419bce9a4

0x468a1ce5593ad1dcdabd3357d31eaa95590e35f1 Nov-19-2021 used HOP swapAndSend 155 USDT
https://polygonscan.com/tx/0xcdc3231aeb9ea0db87a9d29ff8b66a8903c005cb46f669e0077beb22c9133871

0x61dda79dcb2186893bbc6456de302688c84fd978 Nov-19-2021 used HOP swapAndSend 183 USDT
https://polygonscan.com/tx/0x9c06fa7611b1dc80bc99efa6fe1d84733022f90f13dcf05929407d93250795f1

0xc56ee7d507e6b20c867f14916d761d04d0a23450 Dec-11-2021 used HOP swapAndSend 242 USDC
https://polygonscan.com/tx/0x72df49454c2dfd844310fff5f81e8391fee6f0cbac0003c6bf6fffa90d798cdf

0x1791f1e9cf7e7bdcc0147b466b8b9c797db25a74 Nov-19-2021 used HOP swapAndSend 175 USDT
https://polygonscan.com/tx/0x2350978aff5d492a6bcdcf7ffaaab48875431b91f876f3ea48403a4dc1a85da0

0xc46c66313b19189a54cfa5216fa00d7b868a0b50 Nov-19-2021 used HOP swapAndSend 204 USDT
https://polygonscan.com/tx/0x4d0c293a15e737780cb57abc1ec3a2fd24151b6d9773890ed339aa7f60c8818a

0x5df2a53369e688000a726da7fc6fcdbd0a3c5102 Nov-19-2021 used HOP swapAndSend 215 USDT
https://polygonscan.com/tx/0x6d96782d61e95389cd467b69e82888b316887e9a47ddbcfc69b1800ef0a9b40d

0xe4faca92515ea49041f5257eae5fff242c7df385 Nov-19-2021 used HOP swapAndSend 211 USDT
https://polygonscan.com/tx/0xf90f30c3e126441f0ee82c82a580a41d07cabb84a83ca112cd1496363d40aa3b

0x914bc451407f9e27f9914c037e88bf545bec0d8a Dec-11-2021 used HOP swapAndSend 231 USDC
https://polygonscan.com/tx/0x612478c06dd619666bfd3389f495d250cfca1ea00f18de4c4e09938bf4c50e6d

0xeda6de28eef7df0d2c7bba45b9d068e85c2e89e0 Dec-11-2021 used HOP swapAndSend 229 USDC
https://polygonscan.com/tx/0x8b99e5ecc4355480bb324efd97f090736adc0b096dc132e4db03f64eca2619c1

0x6cda151ad03adabdddf8fa15ffbbe7dc1b8071fa Dec-14-2021 used HOP swapAndSend 212 USDC
https://blockscout.com/xdai/mainnet/tx/0x551c447cd4a76cc0baa5eff9700e5ca35b4c15e509b9eed5fae482395455d02a

0xc2fed74f6e44333e1f90becd98c1f41dd31299b7 Dec-14-2021 used HOP swapAndSend 202 USDC
https://blockscout.com/xdai/mainnet/tx/0xfb304f02de27267b7cca7ee0541cf2deb8a8d5abb8bea726c396d21f91674c8b

0x3c32391d71f2bda88941126b2c68fb49a874ccee Dec-14-2021 used HOP swapAndSend 205 USDC
https://blockscout.com/xdai/mainnet/tx/0x1382a10f202a0c9f3400b72d64bc4357cb3a38240fe5793dd88708f1c4260452

0x0d10a6fdf0407580755ed4e9444efa016adf9263 Dec-14-2021 used HOP swapAndSend 233 USDC
https://blockscout.com/xdai/mainnet/tx/0x743c390ac3e9a03cfe8f84ef80ede2186c7e69cae6243eff67f6d01dc288b9f7

0xb1966b52deca0d4b2c9e23977563b948da3b8360 Nov-19-2021 used HOP swapAndSend 2 times total 242 USDT
https://blockscout.com/xdai/mainnet/tx/0x685eab70dba25da802652727b6b01b2fe8ae873ddfe556ee374f2a11eb5e1ed5
https://blockscout.com/xdai/mainnet/tx/0x44bebebc2afc50db8030c452ae20f21698b7af57676d450e8e1f452ca379ff83

0x37980d2ec65676fee74f6a6e34d1125825c0297e Dec-14-2021 used HOP swapAndSend 232 USDC
https://blockscout.com/xdai/mainnet/tx/0x47b77df05bc6488a4bd013c615d641ad976c571bfcbd1ba8263a7a8ce42d74c1

0xfe2b718a9b26be75a67f444619714b8c5d4f6f39 Dec-14-2021 used HOP swapAndSend 238 USDC
https://blockscout.com/xdai/mainnet/tx/0x0d08829db7dda7b8b182489d5c15b981d5dc10f5c30baecbf7ff4ecd2d67f946

0xc8442d72e06cd03e5a5b7a01a4dce99db5540feb Dec-14-2021 used HOP swapAndSend 241 USDC
https://blockscout.com/xdai/mainnet/tx/0xf747a80e802de7b14cfea7c065bf6d755419ae877a2bc2d5196365e329cb2c5e

0x8983d0d5a892865339e48738f77d37ba9be32df7 Dec-14-2021 used HOP swapAndSend 222 USDC
https://blockscout.com/xdai/mainnet/tx/0xaf7a71dc3559eb3c432ac40f5956d73b51d4f43a442efa0d51e7fd8886d41b16

0x7b91b1da16d1700120948eb24aee40f058dff032 Dec-14-2021 used HOP swapAndSend 226 USDC
https://blockscout.com/xdai/mainnet/tx/0x3ace9602b1eb90301a3f3eec3b11bbe078ac5160cee747ca25bb6105a54973b1

0x653e791dcec2d311848742b9ad35ca68acfe4839 Dec-14-2021 used HOP swapAndSend 239 USDC
https://blockscout.com/xdai/mainnet/tx/0x50bfa1c03dc7de344b9ab95d63e564a14f472669820a56252a85c4cd44711a54

# Methodology

I checked polygon's matic transactions with polygonscan API and filtered transactions, where 20+ recipients addresses and sende are eligible for the airdrop. Then I checked when and how this wallets interact with HOP. Node js code available on request.

# Rewards Address

0x7d3f6048091aEdE6198736b8C6ac527c314e25ff

**Comment by shanefontaine** [comment]
Thank you for your report @GreyPK. We have verified that the addresses in this report are Sybil attackers.

The report included 29 eligible addresses as Sybil attackers which means you are eligible for 5151.172909917283014079 HOP!

The qualified addresses are as follows:

```
0x5a49271faa9c068dc409f6ee52cd260971f213b3
0x469748c862cec0a18b6dc10be7484932ba0d84ca
0x8d01fd620ba95d3e38879206b2a8fbec24885b10
0x1c568afbd66f1669b9870cd33c83a6a5de10dd00
0x95dd72e695017c690bdfa7bca821fc2c5df927af
0xa9bbbead5d03e71634870c7bb7e085d8db64e08c
0xe7fc210397d8f654c064adf4712314c7e7c5cd17
0xdbf199ba9617d5f92d9ed0fb28a2f5f7244df836
0x70b62b87c490bd492f788d9e29ed8d367a68b9aa
0x468a1ce5593ad1dcdabd3357d31eaa95590e35f1
0x61dda79dcb2186893bbc6456de302688c84fd978
0xc56ee7d507e6b20c867f14916d761d04d0a23450
0x1791f1e9cf7e7bdcc0147b466b8b9c797db25a74
0xc46c66313b19189a54cfa5216fa00d7b868a0b50
0x5df2a53369e688000a726da7fc6fcdbd0a3c5102
0xe4faca92515ea49041f5257eae5fff242c7df385
0x914bc451407f9e27f9914c037e88bf545bec0d8a
0xeda6de28eef7df0d2c7bba45b9d068e85c2e89e0
0xb1966b52deca0d4b2c9e23977563b948da3b8360
0x6cda151ad03adabdddf8fa15ffbbe7dc1b8071fa
0xc2fed74f6e44333e1f90becd98c1f41dd31299b7
0x3c32391d71f2bda88941126b2c68fb49a874ccee
0x0d10a6fdf0407580755ed4e9444efa016adf9263
0x37980d2ec65676fee74f6a6e34d1125825c0297e
0xfe2b718a9b26be75a67f444619714b8c5d4f6f39
0xc8442d72e06cd03e5a5b7a01a4dce99db5540feb
0x8983d0d5a892865339e48738f77d37ba9be32df7
0x7b91b1da16d1700120948eb24aee40f058dff032
0x653e791dcec2d311848742b9ad35ca68acfe4839
```

**Comment by forweb3** [comment]
sir ，One of my, our report is finding that the rule has to be a jump to survive. However, the successor has completed more than 10 user tracking records, and there is no record of him. I cancelled his reward. And this is what I found so am I not rewarded accordingly? This address has not interacted with hop. And he did not provide other information that he interacted. I report all his reports.  
0x7d3f6048091aEdE6198736b8C6ac527c314e25ff  This is a brand new empty address, why can't he provide the address he interacts with, because he is the real empty hunter. He circumvented it with better technical isolation. So here's my reasoning, I want 25% of his reward to go to me and then the remaining 75% to all eligible real users. Just like our rules.
If the report is illegal, then the judgment should also be cancelled! If not cancelled, then I report this fake report as a sybil attack. His reward should be allocated to me.
@shanefontaine
